pkg/flavors/postgres/postgres.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"fmt"
+
+	valid "github.com/asaskevich/govalidator"
+	"github.com/tetratelabs/getenvoy/pkg/flavors"
+)
+
+// Define template parameter names
+const endpoint string = "endpoint"
+const inport string = "inport"
+
+// Flavor implements flavor.FlavorConfigTemplate interface
+// and stores config data specific to Postgres template.
+type Flavor struct {
+	// Location of the postgres server
+	endpoint string
+	// Envoy's listener port
+	inport string
+}
+
+var flavor Flavor
+
+func init() {
+	// Set default values.
+	// Default values are not required to be present in cmd line.
+	flavor.inport = "5432"
+	flavors.AddTemplate("postgres", flavor)
+}
+
+// CheckParams verifies that passed template arguments are correct and
+// are sufficient for creating a valid config from template.
+func (Flavor) CheckParams(params map[string]string) (interface{}, error) {
+	required := map[string]int{endpoint: 0}
+
+	for param, value := range params {
+		switch param {
+		case endpoint:
+			required[param]++
+			flavor.endpoint = value
+		case inport:
+			if !valid.IsInt(value) {
+				return nil, fmt.Errorf("Value for templateArg %s must be integer number", param)
+			}
+			flavor.inport = value
+		default:
+			fmt.Printf("Ignoring unrecognized template parameter: %s", param)
+		}
+	}
+
+	// Check if all required params have been found in the parameter list
+	var notFound string
+	for key, count := range required {
+		if count == 0 {
+			notFound += key + " "
+		}
+	}
```

Comment:
This is a very generic way to validate parameters.

Especially, when we have only 2 of them :)

May be, we could avoid the `required ` map and `for` loop ?

Reply:
I will see if I can move it to `flavors` package and keep it generic. It could be useful when we expand flavors.

Reply:
Simplified it when I reworked logic to accommodate multiple endpoints.